### ISSUE TITLE
Fix S1118 FN: Support static abstract/virtual interface implementation classes

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6288]
### Description

FN for S1118 (ClassWithOnlyStaticMember).

Using C#11 static abstract/virtual interface methods, the rule is not raising an issue.

### Repro steps

```csharp
public interface IVirtualMethods
{
    virtual static string StaticVirtualMembersInInterfaces(string s1, string s2) => s1 + s2;
    
    virtual static string Prop { get; set; }
}

public class VirtualMethods : IVirtualMethods // FN
{

}

public interface IAbstractMethods
{
    static abstract bool StaticVirtualMembersInInterfaces();
}

public class AbstractMethods : IAbstractMethods // FN
{
    public static bool StaticVirtualMembersInInterfaces() => true;
}


```

### Expected behavior

The rule should raise an issue, since the class only has static members.

### Actual behavior

The rule does not raise an issue.

**Comment by gregory-paidis-sonarsource** [comment]
Since this applies exclusively to classes that implement interfaces and static classes cannot implement interfaces at all, i am closing this issue.
